test/test_extension_point.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2016-2018 Dirk Thomas
+# Copyright 2023 Open Source Robotics Foundation, Inc.
+# Licensed under the Apache License, Version 2.0
+
+import os
+from unittest.mock import DEFAULT
+from unittest.mock import patch
+
+from colcon_core.extension_point import EntryPoint
```

Comment:
TIL Python modules "re-export" their imports.

Is there any concern about tying ourselves to importlib.metadata's EntryPoint class? It's in stdlib so it should be an easier target and have very robust deprecation cycles.

Reply:
I think I've abstracted us from it enough that it shouldn't be a problem. The new API is modeled more around the definition format (entry_points.txt) rather than the API provided.

It's a good sign that I was able to implement both `pkg_resources` and `importlib.metadata` behind the new API successfully.